### PR TITLE
[submission] Yjeong, jkim, wchoi, jchung, sjung

### PR DESCRIPTION
**Pull request by YeongSeokJeong**
**AUTHOR**

Dear mdx workshop committee,

I request a review for the following MDX submission:

### **Title:**  LightSAFT: Lightweight Latent Source Aware Source Frequency Transform for Source Separation

**Author(s):**  

**Category ***

* [ ] `POSTERS` _(+Abstract: 250 Words+one page pdf)_
  > As a participant of the MDX challenge, please know that your poster would be automatically accepted after a minimal prescreening. Hence, don't forget to provide us with the link to your team in the AIcrowd website.
  > You are also welcome to submit us some unrelated abstract, in case your interest would be in presenting some method or idea you would just like to discuss with the participants passing by.  
  > => You *don't* need to include your poster in the PR. Just be ready to have it for D-day and show it to the audience through screen sharing.
* [x] `LONG TALK`
  > A long talk will last about 20-30min, including questions. This can be about some recent paper you published, or may also present some original research and investigations you did. If you pick this category, please provide us with an extended abstract of the talk, or directly with the slides if you have them available, even in an unfinished state, at your convenience. 
* [ ] `DISCUSSIONS`
  > This category is intended to stimulate fruitful discussions and collaborations. If you pick it, it means you are willing to chair some open discussion with the participants, mainly around one particular aspect of music demixing research. For that purpose, you should provide a minimal abstract as well as a list of a few questions you would like the participants to discuss about. Don't hesitate to propose ideas, We will possibly contact you in case we identify several similar topics that were proposed!

**Preferred Timezone***

* [ ] A 11:00 PM (PDT) to 9:00 AM PDT
* [x] B 11:00 AM (PDT) to 9:00 PM PDT

  > The preferred timezone will help us to organize the day schedule. You should select at least one, but can perfectly select both if you don't care. We will organize the programme so that to maximally fit your preferences.
---

**ORGANISATION COMMITTEE (do not fill out)**

* [x] Editor acknowledgment
* [x] Reviewer 1 
* [ ] Reviewer 2
* [x] Review 1 decision [accept/reject]
* [ ] Review 2 decision [accept/reject]
* [ ] Editor decision [accept/reject]

### CODE REVIEW

**Comment by faroit** [comment]
@YeongSeokJeong please also tick the boxes for the submission format and your preferred timezone

**Review by ws-choi — CHANGES_REQUESTED** [review]
typo report

File: paper.md [hunk 1]
```diff
@@ -25,115 +31,175 @@ arxiv-doi: 10.21105/joss.01667
...
+For example, one can easily extend it to sample-based source separation [@lee:2019, @lin:2020].
 
+LaSAFT-Net [@choi:2021] is one of the representative conditioned source separation models.
+It applies the Latent Source aware Frequency Transformation (LaSAFT) blocks.
+A LaSAFT block aims to capture the latent source's frequency patterns depending upon a given symbol that specifies which target source we want to separate.
+Assuming that each latent source contains independent information depending on its viewpoint, it employs the attention mechanism to model the relevance between latent sources and the target symbol.
+With this approach, LaSAFT-Net showed comparable performance to single-source separation models on the MusDB18 [@rafii:2017] benchmark even if conditioned models are usually inferior to single-source separation models.
 
-# Paper structure depending on the category
+However, it consumes numerous parameters to analyze latent sources.
```
Comment (aliutkus) [inline paper.md:57]: * "However, it comes with a very large number of parameters compared to its fixed-source counterpart"
* could you provide some numbers here to illustrate this claim ?

File: paper.md [hunk 2]
```diff
@@ -25,115 +31,175 @@ arxiv-doi: 10.21105/joss.01667
...
+It prevents the model's applicability to a real-world environment, where resources are restricted, such as Music Demix Challenge (MDX) [@mitsufuji:2021] at ISMIR 2021.
```
Comment (aliutkus) [inline paper.md:58]: * I don't really think that MDX did evaluate computing resources as a criterion for evaluation, so this should be changed. 
* However, I totally acknowledge that computing power definitely _is_ a relevant criterion for bringing systems to practical applications. So that actually considering it as a criterion for later occurrences of MDX may be interesting...

File: paper.md [hunk 3]
```diff
@@ -25,115 +31,175 @@ arxiv-doi: 10.21105/joss.01667
...
+To this end, we explore a lightweight version of LaSAFT-Net to make it affordable in a real-world environment by decreasing parameters and simplifying the structure.
```
Comment (aliutkus) [inline paper.md:59]: decreasing the number of parameters

File: paper.md [hunk 4]
```diff
@@ -25,115 +31,175 @@ arxiv-doi: 10.21105/joss.01667
...
-and refer to \autoref{eq:fourier} from text.
+##  Lightweight latent Source Attentive Frequency Transformation Block
 
-## Figures
+The TDF of the original LaSAFT block contains two phases (Figure 1.(a)).
+In the first phase, fully connected layers extract latent sources' features from mixture sources.
+In the second phase, other linear layers are applied to generate the final output.
+Even if each fully connected layers downsample the internal space, this architecture still consumes many parameters.
+It seems that it contains too many connections between two phases because a connection exists for each pair of different latent sources.
+Although such connections might be helpful for separation enabling active communication between sub-modules, it might disturb utilizing the model in the real-time environment.
```
Comment (aliutkus) [inline paper.md:121]: it may be computationally difficult to use this model in real-time environment.

File: paper.md [hunk 5]
```diff
@@ -25,115 +31,175 @@ arxiv-doi: 10.21105/joss.01667
...
 
-Figures can be included like this:
+![Figure 1. Comparison the original LaSAFT block and proposed LightSAFT block.
+The $\mathbb{x}$ is the input intermediate feature and the $\mathbb{v}_i$ is generated latent source.](./img/comparison_of_blocks.png)
 
-![Caption for example figure.](https://raw.githubusercontent.com/mdx-workshop/mdx-workshop.github.io/master/banner.jpg){ width=40% }
+Figure 1 shows the difference between the original LaSAFT block and the proposed LightSAFT block in the latent source separating process.
+The blocks receive the intermediate feature x and generate the latent source V.
+Each FC block comprises a fully connected layer (FC), Batch Norm, and activation function in each block.
+We eliminate inter-source connections between two phases and use a shared linear layer in the phase 2.
```
Comment (aliutkus) [inline paper.md:130]: in phase 2

File: paper.md [hunk 6]
```diff
@@ -25,115 +31,175 @@ arxiv-doi: 10.21105/joss.01667
...
+From this approach, we improve the network's applicability and efficiency.
+To validate that our block is reasonable for the MDX challenge, we evaluated the performance in its environment.
 
-and referenced from text using \autoref{fig:example}.
+## Advanced LightSAFT
 
-## Acknowledgements
+To develop a lightweight LaSAFT-Net, we replace each LaSAFT block with our LightSAFT block.
+We call the revised separation model LightSAFT-Net for the rest of this paper.
+The LightSAFT-Net shows plausible performance with fewer parameters in the MDX challenge.
```
Comment (aliutkus) [inline paper.md:139]: shows good performance, sufficient performance ? (plausible reads a bit weird)

File: paper.md [hunk 7]
```diff
@@ -25,115 +31,175 @@ arxiv-doi: 10.21105/joss.01667
...
+## Advanced LightSAFT
 
-## Acknowledgements
+To develop a lightweight LaSAFT-Net, we replace each LaSAFT block with our LightSAFT block.
+We call the revised separation model LightSAFT-Net for the rest of this paper.
+The LightSAFT-Net shows plausible performance with fewer parameters in the MDX challenge.  
 
-We acknowledge contributions from Brigitta Sipocz, Syrtis Major, and Semyeong
-Oh, and support from Kathryn Johnston during the genesis of this project.
+We further replace each LightSAFT block with a TFC-TDF block in the encoder, to more lighten the number of parameters.
```
Comment (aliutkus) [inline paper.md:141]: tor reduce the number of parameters even more

File: paper.md [hunk 8]
```diff
@@ -25,115 +31,175 @@ arxiv-doi: 10.21105/joss.01667
...
+A TFC-TDF block has fewer parameters than the LightSAFT block because it does not have to generate $|S_L|$ latent source.
+We call this advanced one LightSAFT-Net+.
+Although LightSAFT+ has fewer parameters, it performs better than the existing models in the MDX challenge.
```
Comment (aliutkus) [inline paper.md:144]: you mean better than lasaft ?

File: paper.md [hunk 9]
```diff
@@ -25,115 +31,175 @@ arxiv-doi: 10.21105/joss.01667
...
 
-For a quick reference, the following citation commands can be used:
-- `@author:2001`  ->  "Author et al. (2001)"
-- `[@author:2001]` -> "(Author et al., 2001)"
-- `[@author1:2001; @author2:2001]` -> "(Author1 et al., 2001; Author2 et al., 2002)"
+| model                             | type        | vocals | drums | bass  | other | Avg   |
+| --------------------------------- | ----------- | ------ | ----- | ----- | ----- | ----- |
+| Demucs48-HQ<br/> [@defossez:2019] | Single      | 6.496  | 6.509 | 6.470 | 4.018 | 5.873 |
+| XUMX<br/> [@sawata:2021]          | multi-head  | 6.341  | 5.615 | 5.807 | 3.722 | 5.372 |
+| UMX<br/> [@stoter:2019]           | Single      | 5.042  | 5.357 | 5.504 | 3.309 | 5.042 |
```
Comment (aliutkus) [inline paper.md:170]: did you compute these scores ? The numbers do not match the usual ones found for UMX
see https://github.com/sigsep/open-unmix-pytorch

**Review by aliutkus — COMMENTED** [review]
a method to reduce computational complexity of LaSAFT and still maintain good performance. Cool contribution.

@ethman would you agree to review this ?

File: paper.md (hunk 2 above)
Comment (faroit) [reply paper.md:58]: @aliutkus mdx had limited compute for inference so participants needed to make the model more efficient to get around the compute limitation

File: paper.md (hunk 9 above)
Comment (faroit) [reply paper.md:170]: @aliutkus these numbers do not represent bss_eval and look correct. See https://www.aicrowd.com/challenges/music-demixing-challenge-ismir-2021/leaderboards

File: paper.md (hunk 2 above)
Comment (YeongSeokJeong) [reply paper.md:58]: @faroit Thank you for your explanation! :) In MDX challenge, the participants can not utilize GPU while inference it so the large size models are not available to submit it.

File: paper.md (hunk 3 above)
Comment (YeongSeokJeong) [reply paper.md:59]: @alikus I appreciate finding the typo correction. I will edit it.

File: paper.md (hunk 4 above)
Comment (YeongSeokJeong) [reply paper.md:121]: @aliutkus Thank you for rephrasing it. Your sentence seems to be easier in understanding.

File: paper.md (hunk 5 above)
Comment (YeongSeokJeong) [reply paper.md:130]: @aliutkus I appreciate checking grammatical error. I will edit it.

File: paper.md (hunk 6 above)
Comment (YeongSeokJeong) [reply paper.md:139]: @aliutkus I think sufficient is better for this phrase. I will edit 'plausible' to 'sufficient'.

File: paper.md (hunk 7 above)
Comment (YeongSeokJeong) [reply paper.md:141]: @aliutkus I appreciate finding this error. I will edit it.

File: paper.md (hunk 8 above)
Comment (YeongSeokJeong) [reply paper.md:144]: @aliutkus Our model target to reduce the number of LaSAFT block's parameters. I think that 'the existing models' is uncertain so i will fix it.

File: paper.md (hunk 9 above)
Comment (YeongSeokJeong) [reply paper.md:170]: @aliutkus, @faroit  I apologize mistaken table. The numbers do not precisely pair for UMX and XUMX. I fix it like follows. 

| model                             | type        | vocals | drums | bass  | other | Avg   |
| --------------------------------- | ----------- | ------ | ----- | ----- | ----- | ----- |
| Demucs48-HQ<br/> [@defossez:2019] | Single      | 6.496  | 6.509 | 6.470 | 4.018 | 5.873 |
| XUMX<br/> [@sawata:2021]          | multi-head  | 6.341  | 5.807 | 5.615 | 3.722 | 5.372 |
| UMX<br/> [@stoter:2019]           | Single      | 5.999  | 5.504 | 5.357 | 3.309 | 5.042 |
| LightSAFT                         | conditioned | 6.685  | 5.272 | 5.498 | 4.121 | 5.394 |
| LightSAFT+                        | conditioned | 7.275  | 5.935 | 5.823 | 4.557 | 5.897 |

And i will specify the source.

File: paper.md (hunk 1 above)
Comment (ws-choi) [reply paper.md:58]: - I am sorry for the ambiguity in our wording. What we wanted to point out in that sentence was
   - "However, the existing LaSAFT-Net has a very large number of parameters for latent source analysis." 
   - We wanted to compare LaSAFT-Net <-> LightSAFT-Net, not LaSAFT-Net <-> its fixed-source counterpart.

- Table 1 compares the number of parameters of models. 

| model      | # of parameters | vocals | drums | bass  | other | Avg   |
| ---------- | --------------- | ------ | ----- | ----- | ----- | ----- |
| LaSAFT     | 4.5M            | -      | -     | -     | -     | -     |
| LightSAFT  | 3.8M            | 6.685  | 5.272 | 5.498 | 4.121 | 5.394 |
| LightSAFT+ | 2M              | 7.275  | 5.935 | 5.823 | 4.557 | 5.897 |

**Comment by ethman** [comment]
@aliutkus, sure I can review this. Do I just add inline comments to the markdown file through github? I've written up some notes already. When do you need the review by?

**Comment by aliutkus** [comment]
Cool thanks, yes just inline comments are fine, this is just informal, the objective is just to make the submission better (keep in mind it's accepted anyways it's a small workshop). The sooner the better but it's informal also 😊

File: paper.md [hunk 10]
```diff
@@ -25,115 +31,177 @@ arxiv-doi: 10.21105/joss.01667
...
-It would be nice to have some illustration if applicable.
-
-
-# Example of content fitting the template
-
-## Introduction
+Conditioned source separations have attracted significant attention because of their flexibility, applicability and extensionality.
+Their performance was usually inferior to the existing approaches, such as the single source separation model.
+However, a recently proposed method called LaSAFT-Net has shown that conditioned models can show comparable performance against existing single-source separation models.
+This paper presents Lightsaft-Net, a lightweight version of LaSAFT-Net. As a baseline, it provided a sufficient SDR performance for comparison during the Music Demixing Challenge at ISMIR 2021.
```
Comment (ethman) [inline paper.md:37]: Make sure you standardize the capitalization of LightSAFT, in the abstract it's `lightsaft`, but later in the text it's `LightSAFT`

File: paper.md [hunk 11]
```diff
@@ -25,115 +31,177 @@ arxiv-doi: 10.21105/joss.01667
...
-used in graduate courses on Galactic dynamics to, e.g., provide interactive
-visualizations of textbook material [@Binney:2008]. The combination of speed,
-design, and support for Astropy functionality in `Gala` will enable exciting
-scientific explorations of forthcoming data releases from the *Gaia* mission
-[@gaia] by students and experts alike.
+# Introduction
 
-## Mathematics
+Recently, many methods based on machine learning have been conducted for music source separation.
+They can be distinguished depending upon strategies for separation as follows.: single source separation [@jansson:2017, @choi:2019, @defossez:2019], multi-head source
```
Comment (ethman) [inline paper.md:45]: I understand that this is not the point of this paper, but there's a real opportunity in this first paragraph to create a nice, clear taxonomy of types of source separation systems (this would be a great service to the community as well). To be clear, you don't have to create this taxonomy, but you're already halfway there.

Regardless, as it currently stands there are some issues with clarity and correctness in this paragraph (to be outlined in the nested comments below).

Comment (ethman) [reply paper.md:45]: It would be nice to provide a quick definitions for these different strategies. For instance, I'm not sure what "recursive separation" is without looking up that specific paper. I think there are a few other speech papers that do that. The WHAM! paper comes to mind.

Comment (ethman) [reply paper.md:45]: I think there's a lot in common with the "conditioned source separation" and what you refer to as "sample-based separation" (I would call it Query-by-Example (QBE) separation). Both give the users an additional control input to determine what is being separated, but with the conditioned separation, it's a discrete, one-hot choice, and in the QBE the control input is an audio signal. If I were writing this paper, I would refer to "conditioned separation" as "one-hot conditioned separation", and the audio input separation methods as "Query-by-Example (QBE) separation".

Comment (ethman) [reply paper.md:45]: Here are some other Multi-source separation systems:

- Multi-task U-Net for music source separation by Kadandale et al
- Simultaneous Separation and Transcription of Mixtures with Multiple Polyphonic and Percussive Instruments by Manilow et al.
- Interleaved multitask learning for audio source separation with independent databases by Doire et al.
- It's also important to know that most speech separation systems output multiple speech sources at once

Additionally, there's another class of separation nets that you haven't discussed, which is embedding space separation networks (think Deep Clustering variants). These are also multi-source, but there are also one-hot and QBE conditioning variants (see the latter paper):

- Class-conditional embeddings for music source separation by Seetharaman et al.
- Deep clustering and conventional networks for music separation: Stronger together by Luo et al.

Comment (ethman) [reply paper.md:45]: This is a lot of work, I understand. Given the time constraints, I wouldn't expect all of these changes, but these are things to consider when readying the camera-ready version.

File: paper.md [hunk 12]
```diff
@@ -25,115 +31,177 @@ arxiv-doi: 10.21105/joss.01667
...
+separation [@sawata:2021], conditioned source separation [@cunet:2019, @choi:2021, @samuel:2020, @olga:2021], recursive separation [@recursive:2019], and hierarchical separation [@ethan:2020].
+The conditioned source separation method can separate all sources from the mixture source depending on its condition which determines the target.
```
Comment (ethman) [inline paper.md:47]: I would clarify this sentence a bit. I wouldn't say it can separate *all* sources; just like all of these systems, it's supervised so it's limited by its training data. If you train any of these systems on MUSDB18, none can separate a piano from a guitar. I would reword this so you're not overselling these systems: "A conditioned source separation method separates a target source based on its conditioning input and, at inference time, is able to separate a different source if given a different conditioning input."

File: paper.md [hunk 13]
```diff
@@ -25,115 +31,177 @@ arxiv-doi: 10.21105/joss.01667
...
+It is the one of the most attractive.
```
Comment (ethman) [inline paper.md:48]: This sentence and the next contain a lot of editorializing. I would simplify this to say something like, "in this paper, we are most interested in one-hot conditioned source separation methods due to their flexibility and small parameter count."

File: paper.md [hunk 14]
```diff
@@ -25,115 +31,177 @@ arxiv-doi: 10.21105/joss.01667
...
-and referenced from text using \autoref{fig:example}.
+# Related work
 
-## Acknowledgements
+## TFC-TDF block
 
-We acknowledge contributions from Brigitta Sipocz, Syrtis Major, and Semyeong
-Oh, and support from Kathryn Johnston during the genesis of this project.
+A TFC-TDF [@choi:2019] block consists of a Time-Frequency Convolution (TFC) block and a Time Distributed Fully connected layer (TDF) block.
+A TFC comprises densely connected convolutional blocks containing convolutional neural network layers, Batch Normalization (BN) and ReLU.
```
Comment (ethman) [inline paper.md:80]: I think this explanation of the TFC blocks omits what makes them novel. Many convolutional blocks could be described as "densely connected convolutional blocks containing convolutional neural network layers, Batch Normalization (BN) and ReLU." A more specific description would be great here.

File: paper.md [hunk 15]
```diff
@@ -25,115 +31,177 @@ arxiv-doi: 10.21105/joss.01667
...
+It extracts high-level features from the localized features of the time-frequency dimension.
+A TDF was proposed for Frequency Transformation (FT) [@yin:2020, @choi:2020, @choi:2021], which aims to capture frequency-to-frequency dependencies of the target source.
+@choi:2019 showed that replacing fully convolutional intermediate blocks with TFC-TDF blocks improves separation quality.
+We apply TFC-TDF blocks in the encoder of the original LaSAFT-Net to catch the conditioned source's patterns in an input mixture.
 
-# References
-All submissions should include a reference section.
+## Latent Source
```
Comment (ethman) [inline paper.md:86]: I am very confused by this section. First and foremost, I don't understand what a "latent source" is supposed to be. It is not defined in this section. This section could use substantial clarifications. I feel like this section lacks a clear purpose. It might be good to answer what this section is meant to convey (in simple terms) before you rewrite it.

File: paper.md [hunk 16]
```diff
@@ -25,115 +31,177 @@ arxiv-doi: 10.21105/joss.01667
...
+We present how the original LaSAFT block processes an input feature as follows.
+A LaSAFT block analyzes frequency patterns of latent sources using TDF blocks and aggregates the analyzed patterns by computing an attention matrix between latent sources and a given input conditioning symbol.
+An attention score measures how important the relationship between latent sources and the target is for separation.
+To compute an attention matrix, it first generates a query vector($Q \in \mathbb{R}^{1 \times d_{k}}$), which has $d_{k}$ dimensions, using an embedding layer.
+Key vectors($K \in \mathbb{R}^{|S_L| \times d_{k}}$) are representative vectors of latent sources, which are learnable parameters.
+To compute the attention scores, we apply a dot product of the query and key and take softmax after scaled by $\sqrt{d_k}$.
+
+The multiple TDF blocks generate latent source features $V \in \mathbb{R}^{T \times d_{k} \times |S_L|}$, where the $T$ is the number of frames.
+The attention mechanism which mixes the latent sources' features is as follows:
+$$Attention(Q,K,V')=softmax(QK^{T}/\sqrt{d_{k}})V'$$
```
Comment (ethman) [inline paper.md:114]: Here, you use `V'`, but only `V` is defined in the text above it.

File: paper.md [hunk 17]
```diff
@@ -25,115 +31,177 @@ arxiv-doi: 10.21105/joss.01667
...
+For data augmentation, we generated the mixtures by mixing the different track's sources.  
 
-Citations to entries in paper.bib should be in
-[rMarkdown](http://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html)
-format.
+## Results
+
+| model      | # of parameters | vocals | drums | bass  | other | Avg   |
+| ---------- | --------------- | ------ | ----- | ----- | ----- | ----- |
+| LaSAFT-Net     | 4.5M            | -      | -     | -     | -     | -     |
```
Comment (ethman) [inline paper.md:162]: There are no results for LaSAFT-Net!!!!  Is this intentional? If so, I would argue that including the original LaSAFT results would really strengthen this paper. It answers how much of a tradeoff you make between the full model and the smaller one, which is an important aspect of your paper.

**Review by ethman — COMMENTED** [review]
Overall, really nice work. I think this really rounds out the TFC-TDF/LaSAFT family of architectures! It's quite nice that this latest configuration is also the most lightweight. Congrats on these results; they're quite nice!

Overall, my biggest point of confusion was with the Latent Sources section. I think that, parts of the Introduction, and a few random sentences here and there just need to be spruced up.

Additionally, here are a few overall comments that I couldn't fit within the text:

- Many times throughout the paper Demucs is referred as a single-source system, but in fact it outputs multiple sources in one forward pass.
- There are quite a few issues with citations:
	- In the first paragraph, for Demucs (@defossez:2019), I believe the preferred citation is here: https://github.com/facebookresearch/demucs#how-to-cite 
	- Repeated/doubled citations in the PDF:
		- Audio query-based music source separation
		- A unified model for zero-shot music source separation
		- Conditioned-u-net
		- Meta-tasnet
	- Multiple in-text references to LASAFT-Net point to the AMSS paper (@choi:2021) but they should be @choi2020.
	- In the PDF, there are many in text citations that should be in parenthesis, but are not. Double check to make sure the PDF looks okay with citations. (I'm not sure if this is the issue with the PDF formatter or how you wrote the markdown file. Maybe check with @aliutkus about this).

Overall, great work and I look forward to meeting you at MDX next week!  :-)

File: paper.md (hunk 17 above)
Comment (ws-choi) [reply paper.md:162]: The [AICrowd evaluating system](https://www.aicrowd.com/challenges/music-demixing-challenge-ismir-2021) could not evaluate the original LaSAFT-Net. We tried several times, but the evaluation time exceeded the given threshold defined by the organisers  (approximately 1sec for separating 1sec, I guess) in every trial. Also, the training dataset will be private even after this workshop, as far as we know. Thus, we could not obtain the numbers. Still, we are sorry that we omit numbers without any explanation.

Instead, we can add the evaluation result of the original LaSAFT-Net on MUSDB18 into Table 2 so that readers can compare the performance of the LaSAFT-Net-family.

cc: @YeongSeokJeong

Comment (ethman) [reply paper.md:162]: Oh, yeah, that context is important :-)   I still think these numbers are worth reporting. I would maybe even add a column to this table like "Runnable on MDX system?", and "yes", "no", etc for LightSAFT and LaSAFT? That would drive home the point as well.

File: paper.md (hunk 12 above)
Comment (YeongSeokJeong) [reply paper.md]: We appreciate you taking the time to give feedback on this phrase.

I agree that our insufficient definition of the conditioned source separation method.

So, here's how we paraphrase the sentence: 
- The conditioned source separation method separates target source from a given mixture depending on its conditioning input, which can be various forms (e.g., one-hot conditions, QBE, etc.).

File: paper.md (hunk 13 above)
Comment (YeongSeokJeong) [reply paper.md]: We appreciate your opinion on this paragraph. We want to emphasize the potential of our study via showing the postive aspect of the conditioned source separation method. However, we realize the need to simplify our target.
To make this paragraph more clear, we've rewritten it as follows:
- The conditioned source separation method is still promising because of its flexibility, applicability, and extensionality, despite the fact that it requires more complex mechanisms than the multi-head separation method.
- For example, it is simple to extend it to text-conditioned audio manipulation.
- In this paper, we focus on the one-hot conditioned source separation method since it has extensionality.

File: paper.md (hunk 15 above)
Comment (YeongSeokJeong) [reply paper.md:87]: We appreciate for pointing out crucial topic.
In the original LaSAFT, the latent source is utilized to solve problems caused by symbolically grouped source sets. Our model is based on the original LaSAFT concept.  We think that the motivations and concepts of latent source should be explained in this section.

File: paper.md (hunk 16 above)
Comment (YeongSeokJeong) [reply paper.md]: Thank you for pointing out our typo. I'll edit it `v` to `v'`.

File: paper.md (hunk 10 above)
Comment (YeongSeokJeong) [reply paper.md]: Thank you for catching our typo. We will edit `lightsaft` to `LightSAFT`.

File: paper.md (hunk 11 above)
Comment (YeongSeokJeong) [reply paper.md]: We appreciate your detailed knowledge of source separation tasks. We revised the sentence and citations based-on your suggestions. The revised sentence is as follows: 
- They can be distinguished depending upon strategies for separation as follows.: single source separation [@takahashi:2017; @jansson:2017; @choi:2019], multi-head source separation [@sawata:2021; @kadandale:2020; @manilow:2020; @defossez:2019; @doire:2019], one-hot conditioned separation [@cunet:2019; @choi:2020; @samuel:2020; @olga:2021], Query-by-Example (QBE) separation [@lin:2021; @lee:2019], recursive separation [@wichern:2019; @recursive:2019], embedding space separation [@seetharaman:2019, @luo:2017] and hierarchical separation [@ethan:2020].

**Comment by aliutkus** [comment]
I pick "accept": now that you got plenty of reviews, please finalize updating the paper asap. thanks

**Comment by aliutkus** [comment]
To all participants:
please note that you should update your submission by November 6th 23:00 UTC

Thank you very much

File: paper.md (hunk 17 above)
Comment (ws-choi) [reply paper.md]: Thank you for this comment. We added the column 'runnable?' to make it clear. Also, I agree with the  your suggestion (the Twitter thread you mentioned above). We will include such analysis in our following arXiv paper (after this workshop). Thank you for the comment.

**Comment by faroit** [comment]
hi @YeongSeokJeong. Are you presenting this at the workshop? I can't find you at the registrations

**Comment by faroit** [comment]
👏 The submission has been published on midspace (the ISMIR conference platform). The submissions will also be hosted on the workshop website tomorrow. 

The schedule can be seen there or [here](https://docs.google.com/spreadsheets/d/1PaksFcHWUfNPMg-XloA3CJtrvb9uMU0erCFVyvFi9B4/edit?usp=sharing). Please let us know if there is anything missing. Otherwise we see each other on Friday/Saturday!

**Comment by faroit** [comment]
Hi @YeongSeokJeong @ws-choi , sorry for the long waiting time but I wasn't able to continue working on the proceedings before. 

I managed to finally work on cutting the video recordings of the workshop and plan to upload the recordings to YouTube very soon. In order to allow me to do this, please acknowledge the following:  

> I hereby authorize the right and permission to copyright and/or publish, reproduce or otherwise use my name, voice, and audio-visual recordings. I acknowledge and understand these materials about or of me may be used for both commercial and/or non-commercial purposes.

> I understand that my image may be edited, copied, exhibited, published and/or distributed. There is no time limit on the validity of this release nor are there any geographic limitations on where these materials may be distributed.

> I authorize that the video will be made available under the [CC BY-SA 4.0 license](https://creativecommons.org/licenses/by-sa/4.0/) on the conference website.

and comment on this by replying with `acknowledge`.

**Comment by ws-choi** [comment]
Hi @faroit ,
I **acknowledge** the following:

 > I hereby authorize the right and permission to copyright and/or publish, reproduce or otherwise use my name, voice, and audio-visual recordings. I acknowledge and understand these materials about or of me may be used for both commercial and/or non-commercial purposes.

 > I understand that my image may be edited, copied, exhibited, published and/or distributed. There is no time limit on the validity of this release nor are there any geographic limitations on where these materials may be distributed.

 > I authorize that the video will be made available under the [CC BY-SA 4.0 license](https://creativecommons.org/licenses/by-sa/4.0/) on the conference website.

I sincerely appreciate your hard work.

**Comment by YeongSeokJeong** [comment]
Hi @faroit, 
I acknowledge the following: 

> I hereby authorize the right and permission to copyright and/or publish, reproduce or otherwise use my name, voice, and audio-visual recordings. I acknowledge and understand these materials about or of me may be used for both commercial and/or non-commercial purposes.
> 
> I understand that my image may be edited, copied, exhibited, published and/or distributed. There is no time limit on the validity of this release nor are there any geographic limitations on where these materials may be distributed.
> 
> I authorize that the video will be made available under the [CC BY-SA 4.0 license](https://creativecommons.org/licenses/by-sa/4.0/) on the conference website.
> 

Thank you @faroit.
